spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerPersistentEntityImpl.java

```diff
@@ -163,9 +164,19 @@ public void verify() {
 
 	@Override
 	public String tableName() {
-		return this.tableNameExpression == null
+		return validateTableName(this.tableNameExpression == null
 				? this.tableName
-				: this.tableNameExpression.getValue(this.context, String.class);
+						: this.tableNameExpression.getValue(this.context, String.class));
+	}
+
+	// Because SpEL expressions in table name definitions are allowed, validation is
+	// required.
+	private String validateTableName(String name) {
+		if (Pattern.compile("[^a-zA-Z0-9_]").matcher(name).find()) {
+			throw new SpannerDataException("Only letters, numbers, and underscores are "
+					+ "allowed in table names.");
```

Comment:
Can we also output the table name in the message of the exception to ease debugging?

Reply:
changes made